src/core/components/SlideInBarMenuHeader/SlideInBarMenuHeader.tsx

```diff
@@ -0,0 +1,17 @@
+import { LanguageSwitcher } from '../LanguageSwitcher';
+import ArrowDown from '../../../assets/icons/arrow-down.svg';
+import { SLIDE_IN_MENU_BLOCK_COMPONENT_NAME } from '../../../assets/menu/slideInMenuData';
+
+export interface SlideInBarMenuHeaderProps {
+  title: string;
+  componentName: string | undefined;
+}
+
+export const SlideInBarMenuHeader = ({ title, componentName }: SlideInBarMenuHeaderProps) => {
+  return (
+    <div className="slide-in-bar-menu__nav-header">
+      <span className="slide-in-bar-menu__title">{title}</span>
+      {SLIDE_IN_MENU_BLOCK_COMPONENT_NAME.RIGHT_MENU === componentName && <LanguageSwitcher icon={ArrowDown} isStandardSize={false} />}
```

Comment:
Lets use a  simple boolean property instead of passing so many information into this simple component just to define the visibility of the `LanguageSwitcher  `. The current solution brakes the principle of the single responsibility. For example, if at some point we have to show the `LanguageSwitcher ` in multiple components, then we have to extend the terms and change the body of this component, which should not happen as the component itself didn't change (no visual changes, the structure is same, the functionality is basically the same: show / hide the inner component). But the second responsibility (or dependency on the knowledge about component names) makes us to change the code. Therefore, lets the terms are defined on upper level, and the visibility is controlled by the property.

Reply:
Well noted! Totally agree with you, thank you for suggestion!